src/domu_cfgs.c

```diff
@@ -0,0 +1,224 @@
+		.flags = (XEN_DOMCTL_CDF_hvm | XEN_DOMCTL_CDF_hap | XEN_DOMCTL_CDF_iommu),         \
+                                                                                                   \
+		.tee_type = DT_PROP(node_id, tee),                                                 \
+                                                                                                   \
+		.load_image_bytes = LOAD_IMAGE_BYTES(node_id),                                     \
+		.get_image_size = GET_IMAGE_SIZE(node_id),                                         \
+		.dtb_start = DTB_START(node_id),                                                   \
+		.dtb_end = DTB_END(node_id),                                                       \
+                                                                                                   \
+		.irqs = (uint32_t[])DT_PROP_OR(node_id, domu_irqs, {}),                            \
+		.nr_irqs = DT_PROP_LEN_OR(node_id, domu_irqs, 0),                                  \
+                                                                                                   \
+		.iomems = (struct xen_domain_iomem[]){GET_IOMEMS(node_id)},                        \
+		.nr_iomems = DT_PROP_LEN_OR(node_id, iomems, 0),                                   \
+                                                                                                   \
+		.dtdevs = (char *[])DT_PROP_OR(node_id, dtdevs, {}),                               \
+		.nr_dtdevs = DT_PROP_LEN_OR(node_id, dtdevs, 0),                                   \
+                                                                                                   \
+		.dt_passthrough = (char *[])DT_PROP_OR(node_id, dt_passthrough, {}),               \
+		.nr_dt_passthrough = DT_PROP_LEN_OR(node_id, dt_passthrough, 0),                   \
+                                                                                                   \
+		.back_cfg = {FILL_DOMU_BACK_CFGS(node_id)},                                        \
+	},
+
+static struct xen_domain_cfg domu_cfg[] = {DT_FOREACH_STATUS_OKAY(xen_domu, FILL_DOMU_CFG_ITEM)};
+
+static int domu_create(struct xen_domain_cfg *dom_cfg, uint32_t domid, bool unpause)
+{
+	int ret;
+
+	ret = domain_create(dom_cfg, domid);
+	if (ret) {
+		LOG_ERR("Can't create domain %s", dom_cfg->name);
+		return ret;
+	}
+
+	ret = domain_post_create(dom_cfg, domid);
+	if (ret) {
+		LOG_ERR("Can't post create domain %s", dom_cfg->name);
+		domain_destroy(domid);
+		return ret;
+	}
+
+	if (unpause) {
+		ret = domain_unpause(domid);
+		if (ret) {
+			domain_destroy(domid);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+void create_and_unpause_domus(void)
+{
+	/* make sure that domD will be started first */
+	for (unsigned int i = 0; i < ARRAY_SIZE(domu_cfg); i++) {
+		char *id = strstr(domu_cfg[i].name, "@");
+		uint32_t domid = strtol(id + 1, NULL, 16);
```

Comment:
The `reg` property - is it the same domid?

Reply:
In comments for bindings `domid` is mentioned as `assigned dynamically`, but here is strict dependency on it. May be the meaning should be different - like index or number?

Reply:
> The reg property - is it the same domid?

In my case, it's the same, but we don't have such a field in the domain configuration; we only have a name. However, in the case of a new implementation, I need to store only configurations in DTS without any connection to the particular domain ID. So, this part of code will have another view in the future patches...